misc-scripts/xref_mapping/XrefParser/RefSeqCoordinateParser.pm

```diff
@@ -431,4 +326,107 @@ sub run_script {
+# script:project=>ensembl,host=>ens-staging1,dbname=>homo_sapiens_core_70_37,ofhost=>ens-staging1,...
+# string until : is ignored, hash is built with keys=>values provided
+sub parse_file_string {
+  my ($self, $file_string) = @_;
+
+  $file_string =~ s/\A\w+://xms;
+
+  my @param_pairs = split( /,/xms, $file_string );
+
+  my $params;
+
+  # set provided values
+  foreach my $pair ( @param_pairs ) {
+    my ($key, $value) = split( /=>/mxs, $pair );
+    $params->{$key} = $value;
+  }
+
+  return $params;
+}
+
+# params hash can provide 3 keyed params: exons, check_and_register and overlap
+# exons is the array ref of the exons to process
+# check_and_register may contain a range registry to check_and_register the exons there
+# overlap may contain a range registry to calculate the overlap of the exons there
+# returns the exon_match, which is always 0 if no overlap requested
+sub compute_exons {
+  my ($self, $params) = @_;
+
+  my $exon_match = 0;
+
+  foreach my $exon (@{$params->{exons}}) {
+    if (defined $params->{check_and_register}) {
+      $params->{check_and_register}->check_and_register( 'exon', $exon->seq_region_start, $exon->seq_region_end );
+    }
+    if (defined $params->{overlap}) {
+      my $overlap = $params->{overlap}->overlap_size('exon', $exon->seq_region_start, $exon->seq_region_end);
+      $exon_match += $overlap/($exon->seq_region_end - $exon->seq_region_start + 1);
+    }
+  }
+
+  return $exon_match;
+}
+
+# requires transcript_result and tl_transcript_result hashrefs
+# returns the best_id, best_score and best_tl_score
+sub compute_best_scores {
+  my ($self, $transcript_result, $tl_transcript_result) = @_;
+
+  my $best_score = 0;
+  my $best_tl_score = 0;
+  my $best_id;
+
+  # Comparing the scores based on coding exon overlap
+  # If there is a stale mate, chose best exon overlap score
+  foreach my $tid (sort { $transcript_result->{$b} <=> $transcript_result->{$a} } keys(%{$transcript_result})) {
+    my $score = $transcript_result->{$tid};
+    my $tl_score = $tl_transcript_result->{$tid};
+    if ($score > $TRANSCRIPT_SCORE_THRESHOLD || $tl_score > $TL_TRANSCRIPT_SCORE_THRESHOLD) {
+      if ($tl_score >= $best_tl_score) {
+        if ($tl_score > $best_tl_score) {
+          $best_id = $tid;
+          $best_score = $score;
+          $best_tl_score = $tl_score;
+        } elsif ($tl_score == $best_tl_score) {
+          if ($score > $best_score) {
+            $best_id = $tid;
+            $best_score = $score;
+          }
+        }
+      } elsif ($score >= $best_score) {
+        $best_id = $tid;
+        $best_score = $score;
+      }
+    }
+  }
+
+  return ($best_id, $best_score, $best_tl_score);
+}
+
+# gets the source id for a given refseq acc.
+# requires as args the refseq acc and a hash mapping source names to ids
+# Needs to be moved elsewhere because other parsers benefit from using it
+sub source_id_for_acc {
+  my ($self, $acc, $source_ids) = @_;
+
+  my $source = {
```

Comment:
I'm not at all convinced by this hash creation and disposal. You could at least stop passing source_ids around and stick them in the instance. Barring a print statement, source_ids is only used here. Maybe you could burn the RefSeq prefixes into the source_id cache directly (and earlier), rather than in every ID mapping?